pretix_eth/payment.py

```diff
@@ -186,32 +193,16 @@ def payment_pending_render(self, request: HttpRequest, payment: OrderPayment):
             return template.render(ctx)
 
         wallet_address = WalletAddress.objects.get_for_order_payment(payment).hex_address
-        currency_type = payment.info_data['currency_type']
+        currency_type = request.session['payment_currency_type']
         payment_amount = payment.info_data['amount']
-
         amount_in_ether_or_dai = from_wei(payment_amount, 'ether')
 
-        if currency_type == 'ETH':
-            erc_681_url = f'ethereum:{wallet_address}?value={payment_amount}'
-            amount_manual = f'{amount_in_ether_or_dai} ETH'
-            uniswap_url = f'https://uniswap.exchange/send?exactField=output&exactAmount={amount_in_ether_or_dai}&inputCurrency=0x6b175474e89094c44da98b954eedeac495271d0f&outputCurrency=ETH&recipient={wallet_address}'  # noqa: E501
-        elif currency_type == 'DAI':
-            erc_681_url = f'ethereum:{DAI_MAINNET_ADDRESS}/transfer?address={wallet_address}&uint256={payment_amount}'  # noqa: E501
-            amount_manual = f'{amount_in_ether_or_dai} DAI'
-            uniswap_url = f'https://uniswap.exchange/send?exactField=output&exactAmount={amount_in_ether_or_dai}&outputCurrency=0x6b175474e89094c44da98b954eedeac495271d0f&recipient={wallet_address}'  # noqa: E501
-
-        else:
-            raise ImproperlyConfigured(f'Unrecognized currency: {currency_type}')  # noqa: E501
-
-        web3modal_url = f'https://checkout.web3modal.com/?currency={currency_type}&amount={amount_in_ether_or_dai}&to={wallet_address}'  # noqa: E501
-
-        ctx.update({
-            'erc_681_url': erc_681_url,
-            'uniswap_url': uniswap_url,
-            'web3modal_url': web3modal_url,
-            'amount_manual': amount_manual,
-            'wallet_address': wallet_address,
-        })
+        # Get payment instructions based on the network type:
+        network = request.session['payment_network']
+        network_class: INetwork = networks_dict[network]
+        instructions = network_class.payment_instructions(wallet_address, payment_amount, amount_in_ether_or_dai, currency_type)  # noqa: E501  
```

Comment:
I don't understand this. You already have instances of `INetwork` in the `networks_dict` object. You should just pick it up and use. Also not call it `network_class` of course. Pseudo-code follows:

```python
payment_network = request.session['payment_network']
if payment_network not in networks_dict:
    # FIXME:
    # not sure how exactly it should be handled - but you def should somehow
    # this request can come later and the network type (aka payment_network)
    # does not have to be supported anymore
    raise WrongConfiguration
network = networks_dict[payment_network]
instructions = network.payment_instructions(......)
```

Reply:
I have to change most of this in another feature (adding a "verbose_name" identifier to network classes that are more network friendly. I will make these changes then)!